zarr/n5.py

```diff
@@ -889,13 +919,13 @@ def decode(self, chunk, out=None):
             chunk = self._from_big_endian(chunk)
 
             # read partial chunk
-            if chunk_shape != self.chunk_shape:  # pragma: no cover
+            if chunk_shape != self.chunk_shape:
                 chunk = np.frombuffer(chunk, dtype=self.dtype)
                 chunk = chunk.reshape(chunk_shape)
                 complete_chunk = np.zeros(self.chunk_shape, dtype=self.dtype)
                 target_slices = tuple(slice(0, s) for s in chunk_shape)
                 complete_chunk[target_slices] = chunk
-                chunk = complete_chunk
+                chunk = complete_chunk.tobytes()
```

Comment:
Could you please expand a bit on why this is needed? Mentioning as this effectively doing another allocation/copy here

Reply:
I don't think it's needed, and I removed it